bump_version.exs

```diff
@@ -1,11 +1,61 @@
 #! /usr/bin/env elixir
 
-new_version =
-  "VERSION"
-  |> File.read!()
-  |> String.trim()
-  |> Version.parse!()
-  |> (fn ver -> "#{ver.major}.#{ver.minor}.#{ver.patch + 1}" end).()
-
-File.write("VERSION", new_version)
-IO.puts("Version bumped to #{new_version}")
+defmodule Git do
```

Comment:
Two thoughts : 

1. Do you think this should be converted into a mix task? 

2. Have you  considered replacing the version number in  `mix.exs` vs having `get_latest_version()` in its stead? Rationale behind this approach would be you can check to see if the latest tag is `:gt` than the current version in thee mix file, and then bump if it is. Thoughts?

Reply:
Didn't think about having it as mix task. If we replace version in `mix.exs` file, either it's a manual step or it will require commit from CI. Manual approach, I totally want to avoid as we always miss doing this. And committing from CI kind of feels weird but I am not oppose to it. It will require some more digging though.

Reply:
In Waterpark we used to have the version number in a file, but then when CI bumped the version, it needed to commit a new change, which would kick off a new build, and on into Loopsville.

Reply:
Derp! Bad train of thought on my part... yeah, in retrospect, I wouldn't  suggest it :) Please ignore 2.

Reply:
Actually, the other thing I suggested results in the same problem... I guess that's where I started to go with the last two sentence in the above comment. Rather  than creating tags in ci, prefer to do this as part of a publish operation. Basically, you can take what you got here and make a wrapper task around hex.publish. That if the end goal is around version numbers on hex. 

Maybe it's also I'm not used to incrementing the version (even via a tag) on every commit either,  so the idea is throwing me for a loop 🤔

Reply:
Github action step, I used, It uses commit message convention to upgrade version. See more details here https://github.com/mathieudutour/github-tag-action . We should be matching the same in lib version. Currently yes its only doing patch no increase but based on last commit message, we can update minor or major versions too